pkg/shp/flags/buildrun.go

```diff
@@ -41,9 +44,14 @@ func SanitizeBuildRunSpec(br *buildv1alpha1.BuildRunSpec) {
 	if br == nil {
 		return
 	}
+	if br.BuildRef != nil {
+		if br.BuildRef.Name == "" && br.BuildRef.APIVersion != nil && *br.BuildRef.APIVersion == "" {
+			br.BuildRef = nil
+		}
+	}
 	if br.ServiceAccount != nil {
 		if (br.ServiceAccount.Name == nil || *br.ServiceAccount.Name == "") &&
-			!br.ServiceAccount.Generate {
+			!*br.ServiceAccount.Generate {
```

Comment:
I am not a particular fan of this notation, if other feel the same, we could do a `== false` or similar.

Reply:
Anyway, the linter will report the `== false` style, so keeping it like this.